packages/core/src/RenderPipeline/BasicRenderPipeline.ts

```diff
@@ -166,59 +100,100 @@ export class BasicRenderPipeline {
+    if (independentCanvasEnabled) {
+      const viewport = camera.pixelViewport;
+      const internalColorTarget = PipelineUtils.recreateRenderTargetIfNeeded(
+        engine,
+        this._internalColorTarget,
+        viewport.width,
+        viewport.height,
+        TextureFormat.R8G8B8A8,
+        TextureFormat.Depth24Stencil8,
+        false,
+        false,
+        camera.msaaSamples
+      );
+      const colorTexture = internalColorTarget.getColorTexture(0);
+      colorTexture.wrapModeU = colorTexture.wrapModeV = TextureWrapMode.Clamp;
+      this._internalColorTarget = internalColorTarget;
+    } else {
+      const internalColorTarget = this._internalColorTarget;
+      if (internalColorTarget) {
+        internalColorTarget.destroy();
+        this._internalColorTarget = null;
+      }
     }
+
+    this._drawRenderPass(context, camera, cubeFace, mipLevel, ignoreClear);
   }
 
   private _drawRenderPass(
     context: RenderContext,
-    pass: RenderPass,
     camera: Camera,
     cubeFace?: TextureCubeFace,
     mipLevel?: number,
     ignoreClear?: CameraClearFlags
   ) {
     const cullingResults = this._cullingResults;
     const { opaqueQueue, alphaTestQueue, transparentQueue } = cullingResults;
-    pass.preRender(camera, opaqueQueue, alphaTestQueue, transparentQueue);
-
-    if (pass.enabled) {
-      const { engine, scene } = camera;
-      const { background } = scene;
-      const rhi = engine._hardwareRenderer;
-      const renderTarget = camera.renderTarget || pass.renderTarget;
-      rhi.activeRenderTarget(renderTarget, camera.viewport, mipLevel);
-      renderTarget?._setRenderTargetInfo(cubeFace, mipLevel);
-      const clearFlags = (pass.clearFlags ?? camera.clearFlags) & ~(ignoreClear ?? CameraClearFlags.None);
-      const color = pass.clearColor ?? background.solidColor;
-      if (clearFlags !== CameraClearFlags.None) {
-        rhi.clearRenderTarget(camera.engine, clearFlags, color);
-      }
 
-      if (pass.renderOverride) {
-        pass.render(camera, opaqueQueue, alphaTestQueue, transparentQueue);
-      } else {
-        opaqueQueue.render(camera, pass.mask, PipelineStage.Forward);
-        alphaTestQueue.render(camera, pass.mask, PipelineStage.Forward);
-        if (clearFlags & CameraClearFlags.Color) {
-          if (background.mode === BackgroundMode.Sky) {
-            background.sky._render(context);
-          } else if (background.mode === BackgroundMode.Texture && background.texture) {
-            this._drawBackgroundTexture(engine, background);
-          }
-        }
-        cullingResults.transparentQueue.render(camera, pass.mask, PipelineStage.Forward);
+    const { engine, scene } = camera;
+    const { background } = scene;
+
+    const internalColorTarget = this._internalColorTarget;
+    const rhi = engine._hardwareRenderer;
+    const colorTarget = camera.renderTarget || internalColorTarget;
```

Comment:
you shoud write flipY logic such as`const rt2DEnabled = camera.renderTarget && cubeFace == undefined;` and ` context.applyVirtualCamera(camera._virtualCamera, false);`